### ISSUE TITLE
Issue with JetEngine Localized Date Dynamic Field Callback

### ISSUE THREAD

**Issue by SirLouen** [CrocoBlock/suggestions#2841]
**Describe the bug**
This can be considered both a bug and a feature but from my point of view this is a bug because the Localization feature doesn't do anything different from the standard date callback

When I add a Dynamic Field, with a custom field that returns UNIX Timestamp, I can set a Format Date (localized) callback with the idea of formating the Timestamp to certain format + adjusting to the WordPress selected timezone. 
Otherwise this doesn't make sense (who will ever need a random GMT formated date for a selected unix timestamp?)

**To Reproduce**
1. Create a Dynamic Field
2. Select a custom field from JetEngine that has a UNIX Timestamp
3. Activate "Filter field output"
4. Select  Format Date (localized)

**Expected behavior**
Checking the code, date and date_i18n seem to be the same, since both Callbacks can set a date format. From my point of view Format Date Localized should not only set the format, **but also set the right timezone (timezone_string WP Option)**

**Screenshots**
Not needed

**Desktop (please complete the following information):**
- All

**Smartphone (please complete the following information):**
- All

**Additional context**
After checking JetEngine code, here is the solution:

In the file `wp-content\plugins\jet-engine\includes\components\listings\manager.php`

Create a private function:

```php
function local_date_i18n($format, $timestamp) {
            $timezone_str = get_option('timezone_string') ?: 'UTC';
            $timezone = new \DateTimeZone($timezone_str);
        
            $date = new \DateTime(null, $timezone);
            $date->setTimestamp($timestamp);
            $date_str = $date->format('Y-m-d H:i:s');
        
            $utc_timezone = new \DateTimeZone('UTC');
            $utc_date = new \DateTime($date_str, $utc_timezone);
            $timestamp = $utc_date->getTimestamp();
        
            return date_i18n($format, $timestamp, true);
}
```

Then in the date_i18n case just set this callback:

`$callback = array('Jet_Engine_Listings', 'local_date_i18n');`

```php

case 'date':
		case 'date_i18n':

			if ( ! Jet_Engine_Tools::is_valid_timestamp( $result ) ) {
				$result = strtotime( $result );
			}
			$format = ! empty( $settings['date_format'] ) ? $settings['date_format'] : 'F j, Y';
			$args   = array( $format, $result );
                        $callback = array('Jet_Engine_Listings', 'local_date_i18n');
			break;
```

+Info: https://wordpress.stackexchange.com/a/135049/157160

**Comment by Floydtm** [comment]
We checked everything again and everything seems to be working correctly. However, I will forward your request to the developer. Thanks you!

**Comment by LevelUpGit** [comment]
I have experienced similar issue https://github.com/Crocoblock/suggestions/issues/2373

**Comment by LevelUpGit** [comment]
Hi, it become worse hahahah since daylight saving changed, now query by today for me changes posts late at 02:00 after midnight. 
Jetengine is definitely going by Universal time, and doesn't adjusts to Local time.

![Screenshot_6](https://user-images.githubusercontent.com/69863985/113524536-9665c700-95af-11eb-8bdb-8e92472e14e3.png)

**Comment by LevelUpGit** [comment]
@kaskad88 Hi, is there any update on this issue?

**Comment by futureyoon** [comment]
+1 experiencing the same issue. it doesnt adjust to local time

**Comment by SirLouen** [comment]
@LevelUpGit 
@futureyoon 
they never left it clear, but they updated the localization function and to use wp_date function a filter must be activated
```
add_filter('jet-engine/datetime/use-wp-date', '__return_true', 10, 2);
```

In `functions.php` for example

**Comment by LevelUpGit** [comment]
@SirLouen Thank you, we were searching for solution for a year.

**Comment by LevelUpGit** [comment]
@SirLouen Just tested it further it does include localized time, but filter in query builder and admin columns doesn't work anymore. they still show one day prior.

**Comment by Crocoblock** [comment]
https://github.com/Crocoblock/suggestions/issues/2841#issuecomment-1661966190 - was fixed in JetEngine 3.5.4
